src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,359 @@
+  var: 'vars',
+};
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').Meta | undefined}
+ */
+function createMeta(entry) {
+  const arrify = val => (Array.isArray(val) ? val : [val]);
+  const { added_in, n_api_version, deprecated_in, removed_in, changes } = entry;
+
+  return {
+    changes,
+    added: arrify(added_in ?? []),
+    napiVersion: arrify(n_api_version ?? []),
+    deprecated: arrify(deprecated_in ?? []),
+    removed: arrify(removed_in ?? []),
+  };
+}
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry Section's AST entry
+ * @param {HeadingMetadataParent} head Head node of the entry
+ * @returns {import('../types.d.ts').Section}
+ */
+function createSection(entry, head) {
+  return {
+    textRaw: transformNodesToString(head.children),
+    name: head.data.name,
+    type: head.data.type,
+    meta: createMeta(entry),
+    introduced_in: entry.introduced_in,
+  };
+}
+
+/**
+ * Find name, type, default, desc properties
+ * @param {import('mdast').ListItem} child
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').List}
+ */
+function parseListItem(child, entry) {
+  /**
+   * @type {import('../types.d.ts').List}
+   */
+  const current = {};
+
+  const getRawContent = node => {
+    return entry.rawContent.slice(
+      node.position.start.offset,
+      node.position.end.offset
+    );
+  };
+
+  // Utility to match and extract information based on a regex pattern
+  const extractPattern = (text, pattern, key) => {
+    const [, match] = text.match(pattern) || [];
+    if (match) {
+      current[key] = match.trim().replace(/\.$/, '');
+      return text.replace(pattern, '');
+    }
+    return text;
+  };
+
+  current.textRaw = child.children
+    .filter(node => node.type !== 'list')
+    .map(getRawContent)
+    .join('')
+    .replace(/\s+/g, ' ')
+    .replaceAll(/<!--.*?-->/gs, '');
+
+  let text = current.textRaw;
+
+  // Extract `name`, `type`, `default`, and `desc` fields with helper functions
+  if (RETURN_EXPRESSION.test(text)) {
+    current.name = 'return';
+    text = text.replace(RETURN_EXPRESSION, '');
+  }
+  text = current.name ? text : extractPattern(text, NAME_EXPRESSION, 'name');
+  text = extractPattern(text, TYPE_EXPRESSION, 'type');
+  text = extractPattern(text, DEFAULT_EXPRESSION, 'default');
+
+  // Remove leading hyphens and remaining text becomes `desc`, if any.
+  current.desc = text.replace(LEADING_HYPHEN, '').trim() || undefined;
+
+  // Recursively parse options if available
+  const optionsNode = child.children.find(child => child.type === 'list');
+  if (optionsNode) {
+    current.options = optionsNode.children.map(child =>
+      parseListItem(child, entry)
+    );
+  }
+
+  return current;
+}
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @param {import('../types.d.ts').Section} parentSection
+ */
+function handleEntry(entry, parentSection) {
```

Comment:
The function is 215 lines long 🙇 IMO extracting the functions will help us write tests in the future also it's better for the readability